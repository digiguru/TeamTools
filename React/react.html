<html>
    <head>

           <link rel="stylesheet" href="../Shared/styles.css">

    </head> 
        
    
    <body id="stage" class="coloured-bg">
        <div id="tuckman"></div> 
        <div id="container"></div>
        <div id="comfort"></div>

    </body>
    <script src="../3rdParty/require.min.js" data-main="js/compiled"></script>
</html>