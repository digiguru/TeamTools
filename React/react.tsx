/*
<ul id="users"><li><span class="user">Adam Hall</span><a href="void(0);">X</a></li><li><span class="user">Billie Davey</span><a href="void(0);">X</a></li><li><span class="user">Laura Rowe</span><a href="void(0);">X</a></li><li><span class="user">Simon Dawson</span><a href="void(0);">X</a></li><li><span class="user">Fred</span><a href="void(0);">X</a></li></ul>
*/
class Delete extends React.Component<any, any> {
    render() {
        return <a href="void(0);">X</a>;
    }
}
class User extends React.Component<any, any> {
  render() {
    return <li><span className="user">{this.props.username}</span><Delete /></li>;
  }
}
class UserList extends React.Component<any, any> {
    render() {
        let users = [];

        this.props.users.forEach(function(user) {
            users.push(<User user={user} username={user.name} key={user.name} />);
        });
        return <ul id="users">{users}</ul>;
    }
}
let USERS = [
    {name: "bob"},
    {name: "Donald"}
];
ReactDOM.render(
  <UserList users={USERS} />,
  document.getElementById("container")
);