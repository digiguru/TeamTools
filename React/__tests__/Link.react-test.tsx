
import * as React from 'react';
import Link from '../Link.react';
var renderizer = require('react-test-renderer');

it('Link changes the class when hovered', () => {
  debugger;
  const component = renderizer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});