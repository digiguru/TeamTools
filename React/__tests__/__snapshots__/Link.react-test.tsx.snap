exports[`test Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Facebook
</a>
------------HTML PREVIEW---------------
<a
  class="normal"
  href="http://www.facebook.com"
>
  Facebook
</a>
`;

exports[`test Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Facebook
</a>
------------HTML PREVIEW---------------
<a
  class="hovered"
  href="http://www.facebook.com"
>
  Facebook
</a>
`;

exports[`test Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Facebook
</a>
------------HTML PREVIEW---------------
<a
  class="normal"
  href="http://www.facebook.com"
>
  Facebook
</a>
`;
